flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptivebatch/util/VertexParallelismAndInputInfosDeciderUtils.java

```diff
@@ -658,4 +661,57 @@ public static long calculateDataVolumePerTaskForInput(
             long globalDataVolumePerTask, long inputsGroupBytes, long totalDataBytes) {
         return (long) ((double) inputsGroupBytes / totalDataBytes * globalDataVolumePerTask);
     }
+
+    /**
+     * Logs the data distribution optimization info when a balanced data distribution algorithm is
+     * effectively optimized compared to the num-based data distribution algorithm.
+     *
+     * @param logger The logger instance used for logging output.
+     * @param jobVertexId The id for the job vertex.
+     * @param inputInfo The original input info
+     * @param optimizedJobVertexInputInfo The optimized job vertex input info.
+     */
+    public static void logBalancedDataDistributionOptimizationResult(
+            Logger logger,
+            JobVertexID jobVertexId,
+            BlockingInputInfo inputInfo,
+            JobVertexInputInfo optimizedJobVertexInputInfo) {
+        // Currently, we will not optimize inputs that have two types of correlations at the same
+        // time, skip it.
+        List<ExecutionVertexInputInfo> optimizedExecutionVertexInputInfos =
+                optimizedJobVertexInputInfo.getExecutionVertexInputInfos();
+        int parallelism = optimizedExecutionVertexInputInfos.size();
+        List<ExecutionVertexInputInfo> nonOptimizedExecutionVertexInputInfos =
+                computeNumBasedJobVertexInputInfo(parallelism, inputInfo)
+                        .getExecutionVertexInputInfos();
+        // When the execution vertex input infos of the two are inconsistent, we consider that
+        // data balanced distribution optimization has performed.
+        if (!optimizedExecutionVertexInputInfos.equals(nonOptimizedExecutionVertexInputInfos)) {
+            logger.info(
+                    "Optimized balanced data distribution for vertex[ID:{}] - Details: "
```

Comment:
Optimized the balanced data distribution for vertex {}, which reads from result {} with type number {}